assets/src/edit-story/components/canvas/page.js

```diff
@@ -71,26 +76,50 @@ function Page() {
 		target.style.transform = `translate(${ frame.translate[ 0 ] }px, ${ frame.translate[ 1 ] }px) rotate(${ frame.rotate }deg)`;
 	};
 
+	// Whenever selection change, update moveable rect
+	useEffect( () => {
+		if ( moveable.current ) {
+			moveable.current.updateRect();
+		}
+	}, [ selectedElements ] );
+
+	const selectedElement = selectedElements.length === 1 ? selectedElements[ 0 ] : null;
+	const hasMoveable = selectedElement !== null;
+
+	// This handles single clicks to any selected element.
+	// TODO: proper multi-select
+	// TODO: don't invoke if *any* transformation has happened.
+	const handleSelectionClick = useCallback( ( evt ) => {
+		const clickHandler = getClickHandler( selectedElement.id );
+		if ( clickHandler ) {
+			// Make selection box "transparent", click element and make selection opaque again,
+			evt.persist();
+			targetEl.style.pointerEvents = 'none';
+			clickHandler( evt );
+			targetEl.style.pointerEvents = '';
+		}
+		evt.stopPropagation();
+	}, [ getClickHandler, selectedElement, targetEl ] );
```

Comment:
This is the second part of the messy bit, that we can hopefully skip in the future setup.